core/src/main/java/com/datastax/oss/driver/internal/core/metadata/InitialNodeListRefresh.java

```diff
@@ -62,7 +62,12 @@ public Result compute(
 
     ImmutableMap.Builder<UUID, DefaultNode> newNodesBuilder = ImmutableMap.builder();
 
+    String clusterName = null;
+
     for (NodeInfo nodeInfo : nodeInfos) {
+      if (clusterName == null) {
+        clusterName = nodeInfo.getClusterName();
```

Comment:
The cluster name is also available from `context.getChannelFactory()` (currently package-private but we can add a getter for it): we re-check it on every new connection to ignore foreign nodes if there are stale entries in the peers table ([JAVA-397](https://datastax-oss.atlassian.net/browse/JAVA-397)).

I think it might be easier to grab it from there, this avoids all the impacts on the topology monitor.